Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,407 @@
+// SPDX-FileCopyrightText: 2025 Liamofthesky <157073227+Liamofthesky@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 ReconPangolin <67752926+ReconPangolin@users.noreply.github.com>
+// SPDX-FileCopyrightText: 2025 taydeo <td12233a@gmail.com>
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later AND MIT
+
+using System.Linq;
+using System.Text;
+using Content.Server.Botany.Components;
+using Content.Server.Construction.Completions;
+using Content.Shared._NF.PlantAnalyzer;
+using Content.Shared.Atmos;
+using Content.Shared.Botany.Components;
+using Content.Shared.DoAfter;
+using Content.Shared.Interaction;
+using Content.Shared.PowerCell;
+using Content.Shared.PowerCell.Components;
+using JetBrains.FormatRipper.Elf;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Player;
+using Robust.Shared.Prototypes;
+using static Content.Server.Botany.Components.PlantAnalyzerComponent;
+
+namespace Content.Server.Botany.Systems;
+
+public sealed class PlantAnalyzerSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly PowerCellSystem _cell = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfterSystem = default!;
+    [Dependency] private readonly UserInterfaceSystem _uiSystem = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PlantAnalyzerComponent, AfterInteractEvent>(OnAfterInteract);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerSetMode>(OnModeSelected);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerMutateIterate>(OnMutationIterate);
+        SubscribeLocalEvent<PlantAnalyzerComponent, PlantAnalyzerDeleteDatabankEntry>(OnDeleteDatabaseEntry);
+    }
+
+    private void OnAfterInteract(Entity<PlantAnalyzerComponent> ent, ref AfterInteractEvent args)
+    {
+        if (args.Target == null || !args.CanReach || !_cell.HasActivatableCharge(ent.Owner, user: args.User))
```

Comment:
`args.Target is not {} target`

replace args.Target and args.Target.Value below with just target

Reply:
use the code i put